app/src/main/java/org/simple/clinic/allpatientsinfacility/AllPatientsInFacilityView.kt

```diff
@@ -1,9 +1,65 @@
 package org.simple.clinic.allpatientsinfacility
 
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.widget.FrameLayout
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.jakewharton.rxbinding2.view.RxView
+import io.reactivex.Observable
+import kotlinx.android.synthetic.main.view_allpatientsinfacility.view.*
+import org.simple.clinic.R
+import org.simple.clinic.activity.TheActivity
+import org.simple.clinic.bindUiToController
 import org.simple.clinic.facility.Facility
 import org.simple.clinic.patient.PatientSearchResult
+import org.simple.clinic.widgets.ScreenCreated
+import org.simple.clinic.widgets.ScreenDestroyed
+import org.simple.clinic.widgets.UiEvent
+import javax.inject.Inject
 
-interface AllPatientsInFacilityView {
-  fun showNoPatientsFound(facilityName: String)
-  fun showPatients(facility: Facility, patientSearchResults: List<PatientSearchResult>)
+class AllPatientsInFacilityView(
+    context: Context,
+    attributeSet: AttributeSet
+) : FrameLayout(context, attributeSet), AllPatientsInFacilityUi {
+
+  private val searchResultsAdapter = AllPatientsInFacilityListAdapter()
+
+  @Inject
+  lateinit var controller: AllPatientsInFacilityUiController
+
+  override fun onFinishInflate() {
+    super.onFinishInflate()
+    if (isInEditMode) {
+      return
+    }
+
+    TheActivity.component.inject(this)
+
+    patientsList.apply {
+      setHasFixedSize(true)
+      layoutManager = LinearLayoutManager(context, RecyclerView.VERTICAL, false)
+      adapter = searchResultsAdapter
+    }
+
+    bindUiToController(
+        ui = this,
+        events = Observable.just(ScreenCreated() as UiEvent),
```

Comment:
We don't need this typecast to `UiEvent`, do we?

Reply:
We do, since we get a compilation error because it sees `Observable<ScreenCreated>` instead. This problem goes away if we merge multiple streams,  but if we have only one, we need to explicitly cast it.

Reply:
You could do this instead:
`Observable.just<UiEvent>(ScreenCreated())`

Reply:
I'll try this out!

Reply:
This looks nicer. However, when this opened in the IDE, you might see the `<UiEvent>` part marked as unnecessary by the IDE and be tempted to remove it. This works fine as well.

However, when you try to compile or build via Gradle, this breaks with a type inference problem. The problem is that Kotlin 1.3.40 [ships with a new and improved type inference engine](https://blog.jetbrains.com/kotlin/2019/06/kotlin-1-3-40-released/) which is enabled by default **only** in the IDE and not in Gradle which causes this change in behaviour.

tl;dr Please disable the new type inference engine in the IDE if you've bumped your Kotlin version.